client/sniffer/src/main/java/org/elasticsearch/client/sniff/SnifferBuilder.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.sniff;
+
+import org.elasticsearch.client.RestClient;
+
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Sniffer builder. Helps creating a new {@link Sniffer}.
+ */
+public final class SnifferBuilder {
+    public static final long DEFAULT_SNIFF_INTERVAL = TimeUnit.MINUTES.toMillis(5);
+    public static final long DEFAULT_SNIFF_AFTER_FAILURE_DELAY = TimeUnit.MINUTES.toMillis(1);
+    public static final long DEFAULT_SNIFF_REQUEST_TIMEOUT = TimeUnit.SECONDS.toMillis(1);
+
+    private final RestClient restClient;
+    private long sniffIntervalMillis = DEFAULT_SNIFF_INTERVAL;
+    private long sniffAfterFailureDelayMillis = DEFAULT_SNIFF_AFTER_FAILURE_DELAY;
+    private HostsSniffer hostsSniffer;
+    private long sniffRequestTimeoutMillis = DEFAULT_SNIFF_REQUEST_TIMEOUT;
+    private ElasticssearchHostsSniffer.Scheme scheme = ElasticssearchHostsSniffer.Scheme.HTTP;
+
+    /**
+     * Creates a new builder instance by providing the {@link RestClient} that will be used to communicate with elasticsearch,
```

Comment:
in the past , yes, not anymore. This method had a second argument, leftover, will fix.